Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Tests/Get-ExchangeEmergencyMitigationServiceState.Tests.ps1

```diff
@@ -0,0 +1,192 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingInvokeExpression', '', Justification = 'Pester testing file')]
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidOverwritingBuiltInCmdlets', '', Justification = 'Pester testing file')]
+[CmdletBinding()]
+param()
+BeforeAll {
+    $Script:parentPath = (Split-Path -Parent $PSScriptRoot)
+    $Script:Server = $env:COMPUTERNAME
+    . $Script:parentPath\Get-ExchangeEmergencyMitigationServiceState.ps1
+
+    # We need to overwrite the existing 'Get-Service' function because it doesn't support
+    # the '-ComputerName' parameter on PowerShell 7
+    Function Get-Service {
+        param(
+            [Parameter(Mandatory = $false)]
+            [string]
+            $ComputerName,
+            [Parameter(Mandatory = $false)]
+            [string]
+            $Name
+        )
+    }
+}
+
+Describe "Testing Get-ExchangeEmergencyMitigationServiceState.ps1" {
+
+    Context "Exchange Emergency Mitigation Service Default State" {
+
+        BeforeAll {
+            Mock Invoke-WebRequest -MockWith { return Import-Clixml $Script:parentPath\Tests\InvokeWebRequestEEMS.xml }
```

Comment:
Move the `BeforeAll` action under `Context` to under `Describe` this avoids you having to run the same `Mock` instructions multiple times. Only set a different `Mock` under the `Context` if it is different from the default testing. 

Place the most common `Mock` instruction in the `Describe` and only override it in the `Context` when you need it.

Reply:
Good point. Will do that.